src/platform/Linux/bluez/Helper.cpp

```diff
@@ -1307,26 +1286,104 @@ static void BluezPeripheralObjectsSetup(gpointer apClosure)
     return;
 }
 
-static void BluezOnBusAcquired(GDBusConnection * aConn, const gchar * aName, gpointer apClosure)
+static void BluezOnBusAcquired(GDBusConnection * aConn, const gchar * aName, BluezEndpoint * endpoint)
 {
-    BluezEndpoint * endpoint = static_cast<BluezEndpoint *>(apClosure);
     VerifyOrExit(endpoint != nullptr, ChipLogError(DeviceLayer, "endpoint is NULL in %s", __func__));
 
     ChipLogDetail(DeviceLayer, "TRACE: Bus acquired for name %s", aName);
 
+    endpoint->mpAdapter = nullptr;
```

Comment:
What is something was in the `mpAdapter` before? Looks like a potential memory leak.